pkg/operation/botanist/kubeapiserver_test.go

```diff
@@ -527,30 +516,99 @@ exemptions:
   runtimeClasses: ["random"]
   namespaces: ["random"]
 `),
+									},
 								},
-							},
-						}
+							}
+						})
+
+						It("should add kube-system to exempted namespaces and not touch other fields", func() {
+							var admConfig *admissionapiv1beta1.PodSecurityConfiguration
+
+							admConfig, ok = config.(*admissionapiv1beta1.PodSecurityConfiguration)
+							Expect(ok).To(BeTrue())
+
+							Expect(admConfig.Defaults).To(MatchFields(IgnoreExtras, Fields{
+								"Enforce": Equal("privileged"),
+								// This field is defaulted by kubernetes
+								"Audit":       Equal("privileged"),
+								"Warn":        Equal("baseline"),
+								"WarnVersion": Equal("v1.23"),
+							}))
+							Expect(admConfig.Exemptions.Usernames).To(ContainElement("admin"))
+							Expect(admConfig.Exemptions.Namespaces).To(ContainElements("kube-system", "random"))
+						})
 					})
 
-					It("should add kube-system to exempted namespaces and not touch other fields", func() {
-						Expect(configData).NotTo(BeNil())
+					Context("PodSecurity admission config is v1alpha1", func() {
+						BeforeEach(func() {
+							shootCopy.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins = []gardencorev1beta1.AdmissionPlugin{
+								{
+									Name: "PodSecurity",
+									Config: &runtime.RawExtension{Raw: []byte(`apiVersion: pod-security.admission.config.k8s.io/v1alpha1
+kind: PodSecurityConfiguration
+defaults:
+  enforce: "privileged"
+  enforce-version: "latest"
+  audit-version: "latest"
+  warn: "baseline"
+  warn-version: "v1.22"
+exemptions:
+  usernames: ["admin"]
+  runtimeClasses: ["random"]
+  namespaces: ["random"]
+`),
+									},
+								},
+							}
+						})
+
+						It("should add kube-system to exempted namespaces and not touch other fields", func() {
+							var admConfig *admissionapiv1alpha1.PodSecurityConfiguration
+
+							admConfig, ok = config.(*admissionapiv1alpha1.PodSecurityConfiguration)
+							Expect(ok).To(BeTrue())
+
+							Expect(admConfig.Defaults).To(MatchFields(IgnoreExtras, Fields{
+								"Enforce": Equal("privileged"),
+								// This field is defaulted by kubernetes
+								"Audit":       Equal("privileged"),
+								"Warn":        Equal("baseline"),
+								"WarnVersion": Equal("v1.22"),
+							}))
+							Expect(admConfig.Exemptions.Usernames).To(ContainElement("admin"))
+							Expect(admConfig.Exemptions.Namespaces).To(ContainElements("kube-system", "random"))
+						})
+					})
+				})
+			})
 
-						config, err = runtime.Decode(codec, configData.Raw)
-						Expect(err).NotTo(HaveOccurred())
+			Context("PodSecurity admission config is neither v1alpha1 nor v1beta1", func() {
```

Comment:
I still think that this is awkward as this test is both outside the `Describe("PodSecurity Admission Plugin", func() {` and outside the `Context("When the config is not nil", func() {` scope, and I think it is solvable with a little refactoring. It should at least be under the `Describe("PodSecurity Admission Plugin` scope.